numpy/random/_examples/cython/extending_distributions.pyx

```diff
@@ -106,7 +106,7 @@ def uniforms_ex(bit_generator, Py_ssize_t n, dtype=np.float64):
     # Cast the pointer
     rng = <bitgen_t *> PyCapsule_GetPointer(capsule, capsule_name)
     randoms = np.empty(n, dtype=_dtype)
-    if _dtype is np.float32:
+    if _dtype == np.float32:
```

Comment:
Sorry, I did not look at it careful enough again right now. This is still incorrect for non-native byte order, would need to add an error branch and `elif _dtype == np.float64`. On the up-side, you can just move the above `raise TypeError('Unsupported dtype "%r"' % dtype)` in the else branch then, so that it is nicer anyway.
I can do that if  you are getting tired of this :).

Reply:
done